_aicspylibczi/Reader.cpp

```diff
@@ -407,22 +407,11 @@ namespace pylibczi {
           scale_factor_,
           nullptr);   // use default options
 
-      // TODO how to handle 3 channel BGR image split them as in readSelected or ??? <= split like readSelected
-      auto image = ImageFactory::constructImage(multiTileComposite, &plane_coord_, im_box_, -1);
-      ImageVector imageVector;
-      imageVector.reserve(1);
-      int idx = 0;
-      if (ImageFactory::numberOfChannels(image->pixelType())>1) {
-          auto splitImages = ImageFactory::splitToChannels(image);
-          for_each(splitImages.begin(), splitImages.end(), [&imageVector](Image::ImVec::value_type& image_) {
-              imageVector.push_back(image_);
-          });
-      }
-      else
-          imageVector.push_back(image);
-
-      imageVector.setMosaic(isMosaic());
-      return imageVector;
+      size_t pixels_in_image = m_statistics.boundingBoxLayer0Only.w * m_statistics.boundingBoxLayer0Only.h * bgrScaling;
+      ImageFactory imageFactory(m_pixelType, pixels_in_image);
+      auto image = imageFactory.constructImage(multiTileComposite, &plane_coord_, im_box_, 0, -1);
```

Comment:
again it looks like you don't need the `auto image = ` part of this line.  the `image` var is never being used.